src/library_manager/lib_manager.c

```diff
@@ -451,8 +453,11 @@ static int lib_manager_dma_init(struct lib_manager_dma_ext *dma_ext, uint32_t dm
 
 static int lib_manager_dma_deinit(struct lib_manager_dma_ext *dma_ext, uint32_t dma_id)
 {
-	dma_release_channel(dma_ext->dma->z_dev, dma_id);
-	dma_put(dma_ext->dma);
+	if (dma_ext->dma) {
+		dma_put(dma_ext->dma);
+		if (dma_ext->dma->z_dev)
+			dma_release_channel(dma_ext->dma->z_dev, dma_id);
```

Comment:
I feel like this is an ordering issue -- is dma_ext->dma even valid after dma_put? I missed this in the original commit I think.

Reply:
Yes, I think You are right. Thank you for catching this.

Reply:
yes, should be like on mtl-002
https://github.com/thesofproject/sof/commit/c4f38c2682e20fc08d6697d79862944f3776082b